broker/src/main/java/io/camunda/zeebe/broker/bootstrap/MonitoringServerStep.java

```diff
@@ -45,16 +45,13 @@ void shutdownInternal(
     brokerShutdownContext.removePartitionListener(healthCheckService);
     concurrencyControl.runOnCompletion(
         healthCheckService.closeAsync(),
-        (ok, error) ->
-            forwardExceptions(
-                () -> {
-                  if (error != null) {
-                    shutdownFuture.completeExceptionally(error);
-                  } else {
-                    shutdownFuture.complete(brokerShutdownContext);
-                  }
-                },
-                shutdownFuture));
+        (ok, error) -> {
```

Comment:
:wrench: I think you can simplify this to:

```java
concurrencyControl.runOnCompletion(healthCheckService.closeAsync(), shutdownFuture);
```

Since the `ActorFuture` now also implements `BiConsumer<V, Throwable>`?

Reply:
No it clashes between `ActorFuture<Void>` and `ActorFuture<BrokerStartupContext`